packages/mui-base/src/utils/useAnimationUnmount.ts

```diff
@@ -0,0 +1,55 @@
+'use client';
+import * as React from 'react';
+import { useEventCallback } from './useEventCallback';
+import { ownerWindow } from './owner';
+
+/**
+ * Unmounts the supplied element only if it has no animations or transitions.
+ * @ignore - internal hook.
+ */
+export function useAnimationUnmount(
+  getElement: () => Element | null | undefined,
```

Comment:
Why is this a function and not a ref object?

Reply:
Because of the wrapper in anchored popups, it's the `firstElementChild` of the floating element.

The usage is:
```js
const unmount = useAnimationUnmount(
  () => popupEl?.firstElementChild,
  () => setMounted(false)
);
```